sdk/textanalytics/azure-ai-textanalytics/README.md

```diff
@@ -134,17 +91,54 @@ TextAnalyticsClient textAnalyticsClient = new TextAnalyticsClientBuilder()
+provide an instance of the desired credential type obtained from the [azure-identity][azure_identity] library.
+Note that regional endpoints do not support AAD authentication. Create a [custom subdomain][custom_subdomain] 
+name for your resource in order to use this type of authentication.
+
+Authentication with AAD requires some initial setup:
+* [Install azure-identity][install_azure_identity]
+* [Register a new AAD application][register_AAD_application]
+* [Grant access][grant_access] to Text Analytics by assigning the `"Cognitive Services User"` role to your service principal.
+
+After setup, you can choose which type of [credential][credential_type] from azure.identity to use. 
+As an example, [DefaultAzureCredential][default_azure_credential]
+can be used to authenticate the client:
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables: 
+AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET.
+
+Use the returned token credential to authenticate the client:
+<!-- embedme ./src/samples/java/com/azure/ai/textanalytics/ReadmeSamples.java#L67-L70 -->
+```java
+TextAnalyticsAsyncClient textAnalyticsClient = new TextAnalyticsClientBuilder()
+    .endpoint("{endpoint}")
+    .credential(new DefaultAzureCredentialBuilder().build())
+    .buildAsyncClient();
+```
+
 ## Key concepts
+### Client
+The Text Analytics client library provides a [TextAnalyticsClient][text_analytics_sync_client] and 
+[TextAnalyticsAsyncClient][text_analytics_async_client] to do analysis on batches of documents. It provides both synchronous and
+asynchronous operations to access a specific use of Text Analytics, such as language detection or key phrase extraction.
 
 ### Text Input
-A text input, sometimes called a `document`, is a single unit of input to be analyzed by the predictive models
+A **text input**, sometimes called a **document**, is a single unit of input to be analyzed by the predictive models
 in the Text Analytics service. Operations on Text Analytics client may take a single text input or a collection
-of inputs to be analyzed as a batch.
+of inputs to be analyzed as a batch. 
+See [service limitations][service_input_limitation] for the input, including document length limits, maximum batch size,
+and supported text encoding.
 
 ### Operation Result
 An operation result, such as `AnalyzeSentimentResult`, is the result of a Text Analytics operation, containing a 
 prediction or predictions about a single text input. An operation's result type also may optionally include information
-about the input document and how it was processed.
+about the input document and how it was processed. An operation result contains a `isError` property that allows to
+identify if an operation executed was successful or unsuccessful for the given text input. When the operation results
+an error. You can simply call `getError()` to get `TextAnalyticsError` which contains the reason why it is unsuccessful. 
```

Comment:
"When the operation results an error." I think you meant to use a comma here and include the next sentence, but not sure.

Reply:
yup.